modules/geo/src/internalClusterTest/java/org/opensearch/geo/search/aggregations/bucket/GeoHashGridIT.java

```diff
@@ -34,135 +34,67 @@
-import org.opensearch.geo.GeoModulePluginIntegTestCase;
 import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGrid;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGridAggregationBuilder;
+import org.opensearch.geo.search.aggregations.common.GeoBoundsHelper;
 import org.opensearch.geo.tests.common.AggregationBuilders;
+import org.opensearch.geo.tests.common.RandomGeoGeometryGenerator;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geometry.utils.Geohash;
 import org.opensearch.index.query.GeoBoundingBoxQueryBuilder;
 import org.opensearch.search.aggregations.InternalAggregation;
 import org.opensearch.search.aggregations.bucket.filter.Filter;
 import org.opensearch.test.OpenSearchIntegTestCase;
-import org.opensearch.test.VersionUtils;
 
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Random;
 import java.util.Set;
 
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
-import static org.opensearch.common.xcontent.XContentFactory.jsonBuilder;
 import static org.opensearch.geometry.utils.Geohash.PRECISION;
 import static org.opensearch.geometry.utils.Geohash.stringEncode;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertSearchResponse;
 
 @OpenSearchIntegTestCase.SuiteScopeTestCase
-public class GeoHashGridIT extends GeoModulePluginIntegTestCase {
+public class GeoHashGridIT extends AbstractBucketAggregationIntegTest {
 
-    @Override
-    protected boolean forbidPrivateIndexSettings() {
-        return false;
-    }
+    private static ObjectIntMap<String> expectedDocCountsForGeohash;
 
-    private Version version = VersionUtils.randomIndexCompatibleVersion(random());
+    private static ObjectIntMap<String> multiValuedExpectedDocCountsForGeohash;
 
-    static ObjectIntMap<String> expectedDocCountsForGeoHash = null;
-    static ObjectIntMap<String> multiValuedExpectedDocCountsForGeoHash = null;
-    static int numDocs = 100;
+    private static ObjectIntMap<String> expectedDocCountsForGeoshapeGeohash;
 
-    static String smallestGeoHash = null;
-
-    private static IndexRequestBuilder indexCity(String index, String name, List<String> latLon) throws Exception {
-        XContentBuilder source = jsonBuilder().startObject().field("city", name);
-        if (latLon != null) {
-            source = source.field("location", latLon);
-        }
-        source = source.endObject();
-        return client().prepareIndex(index).setSource(source);
-    }
-
-    private static IndexRequestBuilder indexCity(String index, String name, String latLon) throws Exception {
-        return indexCity(index, name, Arrays.<String>asList(latLon));
-    }
+    private static String SMALLEST_GEO_HASH = null;
```

Comment:
Please be consistent with naming. This should probably be camelCase like the other mutable static fields here.

Reply:
updated.